---
title: "v2.Work.for.Joint.Lab.meeting"
author: "D. Ford Hannum Jr."
date: "10/29/2020"
output: 
        html_document:
                toc: true
                toc_depth: 3
                number_sections: false
                theme: united
                highlight: tango
---

```{r setup, include=TRUE, message=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE)
library(Seurat)
library(ggplot2)
library(data.table)
library(MAST)
library(SingleR)
library(dplyr)
library(tidyr)
library(limma)
library(scRNAseq)
```

# Loading Data

```{r loading data}
wbm <- readRDS('./data/v2/lesser.combined.integrated.NAMED.rds')
```

# Basic Plots

```{r basic plots}

color_pal <- c("#0072B2", "#CC79A7", "#009E73", "#56B4E9","#D55E00",
               "#E69F00","#999999", 'violet',"red", 'black')

DimPlot(wbm, reduction = 'umap', label = T, repel = T, cols = color_pal)
```

# Cluster Naming

```{r}
#head(wbm@meta.data)
wbm$Experiment2 <- ifelse(wbm$Experiment == 'Control', 'Wildtype', wbm$Experiment)
wbm$State2 <- paste0(ifelse(wbm$Condition == 'Enriched','enr_',''), wbm$Experiment2)
#summary(as.factor(wbm$State2))
```

# SingleR

**Reference Datasets**
* *ImmGen Data*: normalized expression values of 830 microarray samples of pure mouse immune cells, generated by the Immunologic Genome Project
* *Mouse RNA Seq Data*: normalized epxression values of 358 bulk RNA-seq samples of sorted cell populations found on GEO


```{r singleR}
# Loading up reference datasets
m.ref.immgen <- ImmGenData()
m.ref.mus <- MouseRNAseqData()

ref <- list(m.ref.immgen, m.ref.mus)
ref.label <- list(m.ref.immgen$label.main, m.ref.mus$label.main)

# Creating a sc experiment from our seurat object

nbeal <- subset(wbm, Experiment2 == 'Wildtype')

DefaultAssay(nbeal) <- 'RNA'

DefaultAssay(wbm) <- 'RNA'

SCnbeal <- as.SingleCellExperiment(nbeal)

SCwbm <- as.SingleCellExperiment(wbm)

# Predicting the Cluster label
pred_cluster <- SingleR(test = SCnbeal,
                        ref = ref,
                        labels = ref.label,
                        method = 'cluster',
                        clusters = SCnbeal$idents)

# Predicting individual cell labels
# pred_cell <- SingleR(test = SCnbeal,
#                      ref = ref,
#                      labels = ref.label,
#                      method = 'single')

## Again for WBM

# Predicting the Cluster label
pred_cluster.wbm <- SingleR(test = SCwbm,
                        ref = ref,
                        labels = ref.label,
                        method = 'cluster',
                        clusters = SCwbm$idents)

# Predicting individual cell labels
# pred_cell.wbm <- SingleR(test = SCwbm,
#                      ref = ref,
#                      labels = ref.label,
#                      method = 'single')
```

### Using all cells

```{r wbm singleR figures}
pred_scores_cluster <- pred_cluster.wbm$scores

# Deleting columns without any values

pred_scores_cluster <- as.data.frame(pred_scores_cluster[,colSums(is.na(pred_scores_cluster)) != nrow(pred_scores_cluster)])

rownames(pred_scores_cluster) <- rownames(pred_cluster.wbm)

colnames(pred_scores_cluster)[8:12] <- paste0(colnames(pred_scores_cluster)[8:12],'-2')

pred_scores_cluster <- gather(pred_scores_cluster, Cell.Type, Score, factor_key = T)

pred_scores_cluster$seurat.cluster <- rep(rownames(pred_cluster.wbm), 12)

pred_scores_cluster$ref <- ifelse(is.na(tstrsplit(pred_scores_cluster$Cell.Type,'-')[[2]]), 'ImmGen','GEO')

pred_scores_cluster$seurat.cluster2 <- as.factor(pred_scores_cluster$seurat.cluster)

pred_scores_cluster$score2 <- ifelse(is.na(pred_scores_cluster$Score), 0, pred_scores_cluster$Score)



ggplot(data = pred_scores_cluster, aes(y = Cell.Type, x = seurat.cluster2, 
                                       fill = ref, alpha = score2)) +
        geom_tile() +
        theme(axis.text.x = element_text(angle = 90, vjust = .2, hjust = .95)) +
        scale_fill_manual(values = c('red','blue'))
```

### Using on Wildtype

```{r nbeal singleR figures}
pred_scores_cluster <- pred_cluster$scores

# Deleting columns without any values

pred_scores_cluster <- as.data.frame(pred_scores_cluster[,colSums(is.na(pred_scores_cluster)) != nrow(pred_scores_cluster)])

rownames(pred_scores_cluster) <- rownames(pred_cluster)

colnames(pred_scores_cluster)[7:12] <- paste0(colnames(pred_scores_cluster)[7:12],'-2')

pred_scores_cluster <- gather(pred_scores_cluster, Cell.Type, Score, factor_key = T)

pred_scores_cluster$seurat.cluster <- rep(rownames(pred_cluster), 12)

pred_scores_cluster$ref <- ifelse(is.na(tstrsplit(pred_scores_cluster$Cell.Type,'-')[[2]]), 'ImmGen','GEO')

pred_scores_cluster$seurat.cluster2 <- as.factor(pred_scores_cluster$seurat.cluster)

pred_scores_cluster$score2 <- ifelse(is.na(pred_scores_cluster$Score), 0, pred_scores_cluster$Score)



ggplot(data = pred_scores_cluster, aes(y = Cell.Type, x = seurat.cluster2, 
                                       fill = ref, alpha = score2)) +
        geom_tile() +
        theme(axis.text.x = element_text(angle = 90, vjust = .2, hjust = .95)) +
        scale_fill_manual(values = c('red','blue'))
```

# Marker Gene Expression

Cell type specific marker gene expression. Genes were added to the list in two different ways: canonical markers that are well known in the field, and genes that distinguished clusters and were found to play a key role in specific cells.

## Literature for Markers: Previously used

**Ighd**: immunoglobulin heavy constant delta. Seems to clearly be expressed by B-cells, but still working on a good reference.

**Gata2**: From Krause paper: a transcription factor required for both lineages but bind in different combinations [ref](https://ashpublications.org/blood/article/113/10/2191/24361/SCL-and-associated-proteins-distinguish-active)

**Cd68**: a human macrophage marker [ref](https://pubmed.ncbi.nlm.nih.gov/7680921/). A more general [ref](https://www.nature.com/articles/labinvest2016116/)
 
**Vcam1**: found papers using Vcam1+ monocytes, but haven't found a great reference.

**Alas2**: an erythroid-specfiic 5-aminolevulinate synthase gene [ref](https://pubmed.ncbi.nlm.nih.gov/12663458/)

**Gata3**: plays a role in the regulation of T-cells [ref](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2998182/)

**Vwf** and **Itga2b**: Markers for megakaryocytes

**Mcpt8** and **Prss34**: mast cell proteases



```{r marker dot plot}
marker.genes <- rev(c('Itga2b','Vwf','Gata3','Alas2','Vcam1','Cd68','Gata2','Ighd'))

DotPlot(wbm, features = marker.genes) +
        ylab ('Cell Cluster') + xlab ('Marker Genes') +
        theme(text = element_text(size = 10, family = 'sans'),
              axis.text.x = element_text(angle = 90, vjust = .2, hjust = .95),
              axis.text.y = element_text(family = 'sans', size = 10),
              axis.title = element_text(family = 'sans', size = 12))

new.markers <- c('Mcpt8','Prss34','Kit','Jchain','Hmgb1', 'Vpreb3','Igkc','Ighm')

DotPlot(wbm, features = new.markers) +
        ylab ('Cell Cluster') + xlab ('Marker Genes') +
        theme(text = element_text(size = 10, family = 'sans'),
              axis.text.x = element_text(angle = 90, vjust = .2, hjust = .95),
              axis.text.y = element_text(family = 'sans', size = 10),
              axis.title = element_text(family = 'sans', size = 12))

```

## Progenitor Markers

![Alt Text](/Users/dfhannum/Desktop/Li_lab/nihms817018f1.jpg)

### HSPCs

Reading through this [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5119546/) it states that in mice all long-term HSCs are Hoxb5+

```{r hoxb5}
VlnPlot(wbm, features = 'Hoxb5', split.by = 'Experiment2')
```

Other markers for HSPCs: Kit, Flt3 (Negative), Ly6a, Cd34, Slamf1.

#### Looking just at wildtype

```{r hspcs}
hsc.genes <- c('Kit', 'Flt3', 'Ly6a', 'Cd34', 'Slamf1')
#hsc.genes %in% rownames(nbeal)

# for (i in hsc.genes){
#         print(VlnPlot(nbeal, i))
# }

VlnPlot(nbeal, features = hsc.genes[1:2], ncol = 2, pt.size = 0)
VlnPlot(nbeal, features = hsc.genes[3:4], ncol = 2, pt.size = 0)
VlnPlot(nbeal, features = hsc.genes[5], ncol = 2, pt.size = 0)

DotPlot(nbeal, features = hsc.genes)
```

#### Looking at all cells

```{r all wbm hspcs}
VlnPlot(wbm, features = hsc.genes[1:2], ncol = 2, pt.size = 0)
VlnPlot(wbm, features = hsc.genes[3:4], ncol = 2, pt.size = 0)
VlnPlot(wbm, features = hsc.genes[5], ncol = 2, pt.size = 0)

DotPlot(wbm, features = hsc.genes)
```


### CMPs

We saw in the SingleR results that CMPs had the highest correlation with stem cells. From the above figure we can see that CMPs show a distinct pattern of cell surface markers: Kit^+^Sca1^-/low^Cd34^+^FcgR^low^


#### Looking at just wildtype

```{r cmps}
cmp.genes <- c('Kit','Ly6a','Cd34','Fcgr1')

VlnPlot(nbeal, features = cmp.genes, ncol = 2, pt.size = 0)

DotPlot(nbeal, features = cmp.genes)
```


* **Kit^+^**: we see all CMPs positive for Kit expression
* **Sca1^-/low^**: we do see some expression of Ly6a but where we do it is low
* **Cd34^+^**: we see high average expression, with some null values
* **FcgR^low^**: we see relatively low to no expression of Fcgr2b, this gene is part of the FcgR surface marker but not sure how well it correlates

**Conclusion**: the identification of CMPs seems pretty spot on


#### Looking at all cells

```{r cmp in all cells}

VlnPlot(wbm, features = cmp.genes, ncol = 2, pt.size = 0)

DotPlot(wbm, features = cmp.genes)

```

# Quantification of Cells 

## By Experiment

```{r plot and graph by experiment}
DimPlot(wbm, reduction = 'umap', split.by = 'Experiment2', cols = color_pal) +
        theme_bw() + ggtitle ('By Experiment')

tbl <- as.data.frame(table(wbm$Experiment2, wbm$idents))

colnames(tbl) <- c('Experiment','Cell Type', 'Count')

tbl$Experiment_count <- NA

for (i in levels(tbl$Experiment)){
        #print(i)
        tbl[tbl$Experiment == i,]$Experiment_count <- sum(tbl[tbl$Experiment ==i,]$Count)
}

tbl$Percentage <- round(tbl$Count/tbl$Experiment_count,4)*100

tbl$Experiment <- factor(tbl$Experiment, levels = levels(tbl$Experiment)[c(3,1,2)])

ggplot(tbl, aes(x = Experiment, y = Percentage, fill = `Cell Type`)) +
        geom_bar(stat = 'identity') + 
        ylim(0,100) +
        scale_fill_manual(values = color_pal) +
        theme_bw() +
        ylab('Percentage of Cells') + xlab('Experiment') +
        #NoLegend()+
        ggtitle('By Experiment') +
        scale_y_continuous(position = 'right') +
        theme(text = element_text(size = 10, family = 'sans'))

ggplot(data = tbl, aes(x = `Cell Type`, y = Experiment)) +
        geom_tile(aes(fill = Percentage)) +
        scale_fill_gradient2(low = 'white', mid = 'red', high = 'darkred',midpoint = 50) +
        geom_text(aes(label = Count)) +
        ggtitle('Experiment by Cluster (Count)') +
        coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = .5, hjust = .5))

ggplot(data = tbl, aes(x = `Cell Type`, y = Experiment)) +
        geom_tile(aes(fill = Percentage)) +
        scale_fill_gradient2(low = 'white', mid = 'red', high = 'darkred',midpoint = 50) +
        geom_text(aes(label = round(Percentage,0))) +
        ggtitle('Experiment by Cluster (Percentage)') +
        coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = .5, hjust = .5))
```

## By State

```{r plot and graph by state}

wbm$State3 <- factor(wbm$State2, levels = levels(as.factor(wbm$State2))[c(6,4,5,3,1,2)])

DimPlot(wbm, reduction = 'umap', split.by = 'State3', cols = color_pal, ncol = 3) +
        theme_bw() + ggtitle ('By State')

tbl <- as.data.frame(table(wbm$State2, wbm$idents))

colnames(tbl) <- c('State','Cell Type', 'Count')

tbl$State_count <- NA

for (i in levels(tbl$State)){
        #print(i)
        tbl[tbl$State == i,]$State_count <- sum(tbl[tbl$State ==i,]$Count)
}

tbl$Percentage <- round(tbl$Count/tbl$State_count,4)*100

tbl$State <- factor(tbl$State, levels = levels(tbl$State)[c(6,3,4,1,5,2)])

ggplot(tbl, aes(x = State, y = Percentage, fill = `Cell Type`)) +
        geom_bar(stat = 'identity') + 
        ylim(0,100) +
        scale_fill_manual(values = color_pal) +
        theme_bw() +
        ylab('Percentage of Cells') + xlab('State') +
        #NoLegend()+
        ggtitle('By State') +
        scale_y_continuous(position = 'right') +
        theme(text = element_text(size = 10, family = 'sans'))

ggplot(data = tbl, aes(x = `Cell Type`, y = State)) +
        geom_tile(aes(fill = Percentage)) +
        scale_fill_gradient2(low = 'white', mid = 'red', high = 'darkred',midpoint = 50) +
        geom_text(aes(label = Count)) +
        ggtitle('States by Cluster (Count)') +
        coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = .5, hjust = .5))

ggplot(data = tbl, aes(x = `Cell Type`, y = State)) +
        geom_tile(aes(fill = Percentage)) +
        scale_fill_gradient2(low = 'white', mid = 'red', high = 'darkred',midpoint = 50) +
        geom_text(aes(label = round(Percentage,0))) +
        ggtitle('State by Cluster (Percentage)') +
        coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = .5, hjust = .5))
```

# Differential Expression

## MEP/Mast Cluster

Comparing Mpl to Migr1

```{r de mep/mcp}
x <- FindMarkers(wbm,
                 ident.1 = 'Mpl',
                 ident.2 = 'Migr1',
                 group.by = 'Experiment',
                 subset.ident = 'MEP/Mast',
                 logfc.threshold = log(1.5),
                 test.use = 'MAST')

x <- x[x$p_val_adj < 0.05,]
x <- x[order(x$avg_logFC, decreasing = T),]


print('Top 6 Up-Regulated DE Genes')
head(x)

print('Top 6 Down-Regulated DE Genes')
tail(x)

VlnPlot(wbm, features = rownames(x)[1:2], split.by = 'Experiment', pt.size = 0) +
        ggtitle('Top 2 Up-regulated Genes')
```

# Subclustering of MEP/Mast Cluster

```{r subcluster mep/mast cells}
prog <- subset(wbm, idents == 'MEP/Mast')

DefaultAssay(prog) <- 'RNA'

prog <- NormalizeData(prog)

prog <- FindVariableFeatures(prog, selection.method = 'vst', nfeatures = 2000)

prog <- ScaleData(prog, features = rownames(prog))

prog <- RunPCA(prog, features = VariableFeatures(prog), verbose = F)

# ElbowPlot(prog)  # going with 10

# DimPlot(prog, reduction = 'pca', group.by = 'Experiment')
# 
# DimPlot(prog, reduction = 'pca', group.by = 'Condition')

# Find what correlates with PC1!!!!!!!!!!!!!

prog <- FindNeighbors(prog, dims = 1:10)

prog <- FindClusters(prog, resolution = 0.6, verbose = F)

prog <- RunUMAP(prog, dims = 1:10, verbose = F)

DimPlot(prog, reduction = 'pca')

DimPlot(prog, reduction = 'umap')

DimPlot(prog, reduction = 'umap', group.by = 'Experiment2')

table(prog$seurat_clusters)

table(prog$seurat_clusters, prog$Experiment2)
```

## Mast Markers in Subclustering

```{r mast markers}
VlnPlot(prog, features = c('Mcpt8','Prss34'), pt.size = 0)
```

Seems like wide expression in all the clusters, with the exception of relatively low expression of Prss34 in subcluster 5 (all Mpl)

## MEP Markers

From the cell surface marker diagram shown earlier MEPs would follow this trend *Kit^+^Ly6a^-^Cd34^-^Fcgr2b^-^*

```{r mep markers}
VlnPlot(prog, features = c('Kit','Ly6a','Cd34','Fcgr2b'), ncol = 2, pt.size = 0)
```

No specific subclustering showing that pattern. Thought once again we see subcluster 5 has the greatest Kit expression.

## Krause Paper Genes

### MEP Genes

```{r mep genes}
mep.genes <- c('Klf1','Fli1','Gata1','Gata2','Runx1','Tal1','Myb')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MEP Genes')
```

### MEP Genes 2

```{r mep2d genes}

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MEP 2 Genes')
```

### CMP Genes

```{r mep2 genes}
mep.genes <- c('Spink2','Csf3r','Ighm','Glipr1','Miat')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]
# 
# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('CMP Genes')
```

### ERP Genes

```{r erp genes}
mep.genes <- c('Rhag','Ahspp','Klf1')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('ERP Genes')
```

### MKP Genes

```{r mkp genes}
mep.genes <- c('Pf4','Vwf','Trpc6','Gp6','Itgb3','Fcer1g','Selp')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MKP Genes')
```
### MEP/ERP

```{r mep-erp genes}
mep.genes <- c('Cnrip1','Ank1','Tfr2','Acsm3','Cpa3')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MEP/ERP Genes')
```

### MEP/MKP

```{r mep-mkp genes}
mep.genes <- c('C3orf58','Pdia5','Rgs18','Pdgfa','Plek','Arhgap6')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]
# 
# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MEP/MKP Genes')
```

### CMP/MEP

```{r mep-cmp genes}
mep.genes <- c('Cd52','Id3','Smim24')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]
# 
# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('CMP/MEP Genes')
```

### MEP-MKP-ERP

```{r mep-erp-mkp genes}
mep.genes <- c('Abcc4','C6orf25','Casp3','Ibtk','Gata1')

# mep.genes %in% rownames(mast.mep)
# rownames(mast.mep)[grepl('Gata', rownames(mast.mep), ignore.case = T)]

# for (i in mep.genes){
#         print(VlnPlot(prog, features = i, pt.size = 0))
# }

DotPlot(prog, features = mep.genes) + coord_flip() + ggtitle ('MEP/MKP/ERP Genes')
```

# Answering Potential Questions from Joint Lab Meeting

```{r question answer}
VlnPlot(wbm, features = 'Cd34', split.by = 'State2')

VlnPlot(prog, features = 'Itga2b', pt.size = 0)
```

